bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandler.java

```diff
@@ -174,4 +175,24 @@
      */
     void handleRemoval();
 
+    /**
+     * This method should return a unique identifier of the device handled by this {@link ThingHandler}.
```

Comment:
> Did you check the example implementation of getDeviceId()in the HueLightHandler or HueBridgeHandler for example?

Of course.

> For the light

... you use the very same content is already present as a [representation property](https://github.com/eclipse/smarthome/pull/3899/files#diff-8fa06d6d05ccc8c9a1724a0b84c61e5fR119).

> for the bridge

... you use the MAC address, which we specifically mention as a candidate for a representation property.

> because these properties could be set manually during creation of that thing

Afaik, it is not possible to set properties manually, only config parameter values.
But in any case: What you might not know is that ThingHandlers can update the properties at any time (irrespectively whether the Thing is discovered, defined manually (in database or DSL) and this is what the handlers are doing already all the time. So once your handler establishes the connection to the device, it should update the Thing properties. The nice side effect is that in case your thing is stored in the managed provider the updated properties are even persisted. So you will have it at hand after a restart even if your Thing hasn't come online yet.